src/test/java/org/sagebionetworks/bridge/reporter/worker/RetentionReportGeneratorTest.java

```diff
@@ -82,9 +89,41 @@ public void testStudyStartDate() throws Exception {
         
         Map<String, List<Integer>> map = (Map<String, List<Integer>>) report.getData();
         assertEquals(map.get("bySignIn").get(4), new Integer(1));
-        assertEquals(map.get("byUploadedOn").get(4), new Integer(1)); 
+        assertEquals(map.get("byUploadedOn").get(3), new Integer(1)); 
+        
+        verify(bridgeHelper).getAllAccountSummaries(STUDY_ID);
+        verify(bridgeHelper).getStudyPartcipant(STUDY_ID, USER_ID_1);
+        verify(bridgeHelper).getActivityEventForParticipant(STUDY_ID, accountSummary.getId());
+        verify(bridgeHelper).getRequestInfoForParticipant(STUDY_ID, accountSummary.getId());
+    }
+    
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testSignInOnInDifferentTimeZone() throws Exception {
+        AccountSummary accountSummary = mockAccountSummary(USER_ID_1);
+        List<AccountSummary> accountSummaries = new ArrayList<>();
+        accountSummaries.add(accountSummary);
+        Iterator<AccountSummary> accountSummaryIter = accountSummaries.iterator();
+        when(bridgeHelper.getAllAccountSummaries(STUDY_ID)).thenReturn(accountSummaryIter);
+        
+        mockStudyParticipant(bridgeHelper, USER_ID_1, new ArrayList<>());
+        
+        mockActivityEventList(accountSummary, mockStudyStateDateEvent(STUDY_START_DATE_1));
+        
+        RequestInfo requestInfo = mockRequestInfo(SIGN_IN_ON_TIMEZONE, UPLOADED_ON);
```

Comment:
We should also have UPLOADED_ON with timezone, to make sure that we have both cases covered.